apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/CommonQueries.java

```diff
@@ -0,0 +1,211 @@
+   * @param searchType used to denote what the searchValue represents; values are: mbi, mbi-hash,
+   *     hich-hash.
+   * @param searchValue identifier value used to search for a BENE_ID.
+   * @param callerClassName class name of caller; used metrics tracking.
+   * @return long BENE_ID for a beneficiary or zero if not found.
+   */
+  public static long findBeneficiaryIdentifier(
+      EntityManager entityManager,
+      MetricRegistry metricRegistry,
+      String searchType,
+      String searchValue,
+      String callerClassName) {
+
+    Timer.Context beneHistoryTimer =
+        CommonTransformerUtils.createMetricsTimer(
+            metricRegistry, callerClassName, "query", "bene_by_mbi", "bene_by_mbi_or_id");
+    /*
+     * the function returns a String with comma-delimited values; done this way to avoid having
+     * to register a Hibernate custom data type handler for an Array of bigint values.
+     */
+    List<Long> values = null;
+    try {
+      List<Object> rawValues =
+          entityManager
+              .createNativeQuery(FIND_BENE_ID_FROM_IDENTIFIER_SQL)
+              .setParameter("searchIdType", searchType)
+              .setParameter("searchIdValue", searchValue)
+              .getResultList();
+
+      // the db function returns a string that ostensibly could contain
+      // comma-seperated BENE_ID values due to the possibility of 'hash
+      // collisions' for either MBI or HICN. This anomaly will eventually
+      // go away and the function will simply return a long (bigint) when
+      // support for hashed MBI and hashed HICN lookups are retired. At that
+      // point only actual MBI value lookups will be supported and if we have
+      // more than one BENE_ID per MBI_NUM then Medicare has a BIG problem!
+
+      if (rawValues != null && rawValues.size() > 0 && rawValues.get(0) != null) {
+        String arrayVals = (String) rawValues.get(0);
+        values =
+            Arrays.asList(arrayVals.split(",")).stream()
+                .map(String::trim)
+                .map(Long::parseLong)
+                .collect(Collectors.toList());
+      }
+    } catch (Exception e) {
+      throw e;
+    } finally {
+      long queryNanoSeconds = beneHistoryTimer.stop();
+      CommonTransformerUtils.recordQueryInMdc(
+          "bene_by_mbi.bene_by_mbi_or_id", queryNanoSeconds, values != null ? values.size() : 0);
+    }
+
+    if (values == null || values.size() < 1) {
+      throw new NoResultException("Failed to find BENE_ID for: " + searchType);
+    } else if (values.size() > 1) {
+      BfdMDC.put(
+          "database_query_by_hash_collision_distinct_bene_ids", Long.toString(values.size()));
+      throw new UnclassifiedServerFailureException(
+          404,
+          "By hash query found more than one distinct BENE_ID: "
+              + values.size()
+              + ", DistinctBeneIdsList: "
+              + values);
+    }
+    return values.get(0);
```

Comment:
probably can't do that; `List<Long>` values is out of scope by the time we check to see what we can return.

Reply:
also, List<String> rawValues = ... will cause rawValues to be null; the JDBC layer will return a ResultSet of Object(s)...JPA will not perform inline cast just because you specified a List<String>.